oml/functional/metrics.py

```diff
@@ -107,6 +109,9 @@ def calc_retrieval_metrics(
         mean_ap = torch.sum((correct_preds / positions) * gt_tops[:, :k_map], dim=1) / n_gt_matrix
         metrics["map"][k_map_show] = mean_ap
 
+    for fmr_val in fmr_vals:
+        metrics["fnmr@fmr"][fmr_val] = calc_fnmr_at_fmr(distances, mask_gt, fmr_val)
```

Comment:
We have a problem here with the logic of applying `mask_to_ignore` (we want some of the samples in the gallery don't affect the final metric, for existing metrics it may be done via masking the corresponding distances with inf values, check the code)

For the proposed metric this approach does not help since the quantile function does not know that some of the samples in the gallery have to be ignored. But NumPy docs say:
```
Compute the qth quantile of the data along the specified axis, while ignoring nan values.
```
As a solution, I can suggest costing infs to nans inside your function, but perhaps you may suggest a better way...

@DaloroAT what do you think?

Reply:
@dapladoc Do you understand the problem? If not, please, ask us questions

Reply:
Yes, I understand the problem. I'll take into account `mask_to_ignore` not to consider these elements for the metric evaluation.